models/opensea/ethereum/opensea_v4_ethereum_trades.sql

```diff
@@ -27,526 +25,518 @@ with source_ethereum_transactions as (
     from {{ source('ethereum','transactions') }}
     {% if not is_incremental() %}
-    where block_time >= date '{{c_seaport_first_date}}'  -- seaport first txn
+        where block_time >= date '{{c_seaport_first_date}}'  -- seaport first txn
     {% endif %}
     {% if is_incremental() %}
-    where block_time >= date_trunc("day", now() - interval '1 week')
+        where block_time >= date_trunc("day", now() - interval '1 week')
     {% endif %}
 )
-,ref_tokens_nft as (
-    select *
-    from {{ ref('tokens_nft') }}
-    where blockchain = 'ethereum'
+
+, ref_tokens_nft as (
+    select * from {{ ref('tokens_nft') }} where blockchain = 'ethereum'
 )
-,ref_tokens_erc20 as (
-    select *
-    from {{ ref('tokens_erc20') }}
-    where blockchain = 'ethereum'
+
+, ref_tokens_erc20 as (
+    select * from {{ ref('tokens_erc20') }} where blockchain = 'ethereum'
 )
-,ref_nft_aggregators as (
-    select *
-    from {{ ref('nft_aggregators') }}
-    where blockchain = 'ethereum'
+
+, ref_nft_aggregators as (
+    select * from {{ ref('nft_aggregators') }} where blockchain = 'ethereum'
 )
-,ref_nft_aggregators_marks as (
-    select *
-    from {{ ref('nft_ethereum_aggregators_markers') }}
+
+, ref_nft_aggregators_marks as (
+    select * from {{ ref('nft_ethereum_aggregators_markers') }}
 )
-,source_prices_usd as (
+
+, source_prices_usd as (
     select *
     from {{ source('prices', 'usd') }}
     where blockchain = 'ethereum'
     {% if not is_incremental() %}
-      and minute >= date '{{c_seaport_first_date}}'  -- seaport first txn
+        and minute >= date '{{c_seaport_first_date}}'  -- seaport first txn
     {% endif %}
     {% if is_incremental() %}
-      and minute >= date_trunc("day", now() - interval '1 week')
+        and minute >= date_trunc("day", now() - interval '1 week')
     {% endif %}
 )
-,iv_orders_matched AS (
-    select evt_block_time as om_block_time
-          ,evt_tx_hash as om_tx_hash
-          ,evt_index as om_evt_index
-          ,posexplode(orderhashes) as (om_order_id, om_order_hash)
-      from {{ source('seaport_ethereum','Seaport_evt_OrdersMatched') }}
-     where contract_address = '{{c_seaport_contract_address}}'
+
+, iv_orders_matched AS (
+    select 
+          evt_block_time as om_block_time
+        , evt_tx_hash as om_tx_hash
+        , evt_index as om_evt_index
+        , posexplode(orderhashes) as (om_order_id, om_order_hash)
+    from {{ source('seaport_ethereum','Seaport_evt_OrdersMatched') }}
+    where contract_address = '{{c_seaport_contract_address}}'
+)
+
+, iv_orders_fulfilled AS (
+    select 
+          consideration
+        , contract_address
+        , evt_block_number
+        , evt_block_time
+        , evt_index
+        , evt_tx_hash
+        , offer
+        , offerer
+        , recipient
+        , zone
+        , orderHash AS order_hash
+        , posexplode(offer) as (offer_idx, offer_item)
+        , posexplode(consideration) as (consideration_idx, consideration_item)
```

Comment:
Hi @DevCav,
The current engine only allows one `posexplode()` per selection.
We may need to remove one posexplode here(maybe we can write a CTE for posexplode).